docs/content/getting-started/ensign.md

```diff
@@ -88,43 +99,69 @@ data["timestamp"] = time.Now().String()
 data["message"] = "Let's get this started!"
 ```
 
-Next, we will convert our map into an event, which will allow you to specify the mimetype of the message you intend to send (in this case, we'll say it's JSON), and the event type (which will be a generic event for this example). You'll also need to pass in a `TopicId`, which will be a string. If you aren't sure what `TopicId` to use, you can quickly [log into your Ensign dashboard](https://rotational.io/ensign/) and look it up. For this example, we'll pretend it's `"quality-lemon-time"`:
+Next, we will convert our map into an event, which will allow you to specify the mimetype of the message you intend to send (in this case, we'll say it's JSON), and the event type (which will be a generic event for this example).
 
 ```golang
-e := &api.Event{
-    TopicId:  "quality-lemon-time",
+e := &ensign.Event{
     Mimetype: mimetype.ApplicationJSON,
     Type: &api.Type{
-        Name:    "Generic",
-        Version: 1,
+		Name:         "Generic",
+		MajorVersion: 1,
+		MinorVersion: 0,
+		PatchVersion: 0,
     },
 }
 ```
 
-Next, we'll marshal our dictionary into the `Data` attribute of our sample event, and publish it by calling the `Publish` method on the publisher we created above:
+Next, we'll marshal our dictionary into the `Data` attribute of our sample event
 
 ```golang
-e.Data, _ = json.Marshal(data)
-pub.Publish(e)
+if e.Data, err = json.Marshal(data); err != nil {
+    panic("could not marshal data to JSON: " + err.Error())
+}
 ```
 
+### Publish Your Event
+
+Now we can publish your event by calling the `Publish` method on the Ensign client we created above. You'll also need to pass in a `TopicId`, which will be a string. If you aren't sure what `TopicId` to use, you can quickly [log into your Ensign dashboard](https://rotational.io/ensign/) and look it up.
+
+But the truth is that it's hard for humans to remember UUID and you have enough on your plate already. So, you can also use the name of the topic instead of the id. For this example, we'll pretend we want to publish to a topic named `"quality-lemon-time"`.
```

Comment:
```suggestion
But the truth is that it's hard for humans to remember ULID and you have enough on your plate already. So, you can also use the name of the topic instead of the id. For this example, we'll pretend we want to publish to a topic named `"quality-lemon-time"`.
```